### ISSUE TITLE
Differing array argument names in forward and public declarations cause a compiler crash

### ISSUE THREAD

**Issue by neiam00** [Zeex/pawn#637]
### Issue description:
When declaring a public function, we can use arrays as arguments:
```pawn
forward PublicFunction(myarray[]);
public PublicFunction(myarray[]) {}
// Or
forward PublicFunction(const myarray[]);
public PublicFunction(const myarray[]) {}
```
However, if we use a different argument name in the actual public function declaration:
```pawn
forward PublicFunction(const myarray[]);
public PublicFunction(const whatever[])
{
    //
}
```
The compiler should raise error 025: `function heading differs from prototype`. Instead, it crashes and outputs nothing.

### Minimal complete verifiable example (MCVE):
```pawn
forward PublicFunction(const a[]);
public PublicFunction(const b[])
{
  return 0;
}
```

### Workspace Information:

* Compiler version: 3.10.10 (cannot reproduce in 3.2.3664, expected behavior happens)
* Command line arguments provided (or sampctl version): `-d0`, `-O1`, `-d0 -O1`, none
* Operating System: Windows 10

**Comment by Y-Less** [comment]
It shouldn't crash, it really shouldn't do anything.  Names shouldn't matter at all.

**Comment by Daniel-Cortez** [comment]
Already fixed in #561.
